contracts/MIPS.sol

```diff
@@ -46,24 +46,22 @@ contract MIPS {
       bytes32 newStateHash = m.WriteMemory(stateHash, addr, value);
       require(m.ReadMemory(newStateHash, addr) == value, "memory readback check failed");
       return newStateHash;
-    } else {
-      assembly {
-        // TODO: this is actually doing an SLOAD first
-        sstore(addr, value)
-      }
-      return stateHash;
     }
+    assembly {
+      // TODO: this is actually doing an SLOAD first
+      sstore(addr, value)
+    }
+    return stateHash;
   }
 
   function ReadMemory(bytes32 stateHash, uint32 addr) internal returns (uint32 ret) {
     if (address(m) != address(0)) {
       emit TryReadMemory(addr);
       ret = m.ReadMemory(stateHash, addr);
       //emit DidReadMemory(addr, ret);
-    } else {
-      assembly {
-        ret := sload(addr)
-      }
+    }
+    assembly {
+      ret := sload(addr)
```

Comment:
Are you sure this one is right? The if block doesn't return so now the else block is always triggered

Reply:
i think returning early costs same amount of gas as else statement, can double check if needed

Reply:
Would be interested in the check, as the if else is otherwise a bit cleaner in this case imho.

Reply:
![image](https://user-images.githubusercontent.com/4612986/157755580-0fb4c27d-cc85-44dc-9537-ffc1eda12bf6.png)

:point_up: that's the check

tbh though I'd be happy to revert this change I kinda agree the else is cleaner here

whatever you think is best

Reply:
Spent two minutes thinking about this, and really, it doesn't matter at all, they're both pretty simple code paths (and same cost, thanks for checking). I will merge as-is. Thanks for the patch! 🙏